lib/pay/paddle/billable.rb

```diff
@@ -56,7 +56,19 @@ def charge(amount, options = {})
           effective_from: "immediately"
         )
 
-        # This charge is then created from the webhook
+        transaction = ::Paddle::Transaction.list(subscription_id: subscription.processor_id).data.try(:first)
```

Comment:
Creating a one-time charge required a subscription in Paddle Classic. I believe Paddle billing lets you create a one-time charge outside of a subscription now.

https://developer.paddle.com/api-reference/transactions/create-transaction

If so, we should change this to be consistent with the other integrations we have.

Reply:
Transactions can be created, but they don't seem to charge on creation even if `collection_mode` is set to `automatic` so I'm not sure if that's a bug on Paddle's end or if that is intended :thinking:

Reply:
Their API is awfully confusing sometimes. 🙃

Reply:
Just noticed the reply I had from Kieran, head of dev experience at Paddle - https://twitter.com/k__mountford/status/1714729918326534235

Payment details are tied to a subscription, so this is the only way at the moment. They say payment methods at a customer level will be coming later, which would mean this could be changed

Reply:
That's good news. In the meantime, maybe we should still switch this so it generates a Transaction and document that it needs to be passed to Checkout.

It seems far more common that people want to make one-time charges that don't require a subscription. Then we can improve this further when that feature rolls out.